Formula/plotjuggler.rb

```diff
@@ -0,0 +1,35 @@
+class Plotjuggler < Formula
+  desc "Visualize time series that is fast, powerful and intuitive"
+  homepage "https://github.com/facontidavide/PlotJuggler"
+  url "https://github.com/facontidavide/PlotJuggler.git", tag: "3.7.1", revision: "fb2bab446f63c726f85e71b155d9fcbeb5603dc1"
+  license "MPL-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "freetype"
+  depends_on "gettext"
+  depends_on "giflib"
+  depends_on "glib"
+  depends_on "jpeg-turbo"
+  depends_on "libpng"
+  depends_on "libtiff"
+  depends_on "lz4"
+  depends_on "mosquitto"
+  depends_on "pcre2"
+  depends_on "protobuf@21"
+  depends_on "qt@5"
+  depends_on "webp"
+  depends_on "xz"
+  depends_on "zeromq"
+  depends_on "zstd"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", "-DCMAKE_INSTALL_PREFIX=install"
+    system "cmake", "--build", "build", "--config", "RelWithDebInfo", "--parallel", "--target", "install"
+    libexec.install Dir["install/bin/*"]
+    bin.install_symlink libexec/"plotjuggler"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/plotjuggler --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thank you for your comment.
Formula is a GUI program and I can't find the ability to convert or file output on the command line. I am thinking of switching to it because it has the ability to load test data into the GUI. As it just loads the data, I can't mechanically check that it has loaded, but do you have any ideas?